rasa/core/agent.py

```diff
@@ -845,15 +857,16 @@ def _create_domain(domain: Union[Domain, Text]) -> Domain:
                 "type '{}' with value '{}'".format(type(domain), domain)
             )
 
-    @staticmethod
```

Comment:
Some thoughts about this: by making this an instance method, we're changing the public interface of `Agent`, right? `Agent.create_tracker_store()` would stop working, in this case.

However, I'm not sure how we're handling changes to the public Rasa Python API and version numbers. Are we OK in doing these changes through minor version bumps (`1.4.X` to `.1.5.X`)?

Last thought: I noticed that the function wasn't documented in https://rasa.com/docs/rasa/api/agent/ (maybe because it's static?). If it's going to be a public instance method I think we should add a docstring so that it gets rendered in those docs.

Reply:
reverted it 👍 (it's now static again)